xarray/core/dataset.py

```diff
@@ -8696,13 +8735,21 @@ def _wrapper(Y, *coords_, **kwargs):
             else:
                 name = f"{str(name)}_"
 
+            input_core_dims = [reduce_dims_ for _ in range(n_coords + 1)]
```

Comment:
I factored this out of the function call because Black was suggesting something like
```python
apply_ufunc(
    ...,
    input_core_dims=[reduce_dims_ for _ in range(n_coords + 1)]
    + [
        [] for _ in range(3 * n_params)
    ],  # core_dims for p0 and bounds
    ...
)
```

which I found quite ugly.